features/messages/impl/src/test/kotlin/io/element/android/features/messages/media/viewer/MediaViewerPresenterTest.kt

```diff
@@ -90,26 +91,24 @@ class MediaViewerPresenterTest {
             state.eventSink(MediaViewerEvents.SaveOnDisk)
             state = awaitItem()
             assertThat(state.snackbarMessage).isNotNull()
-            state = awaitItem()
-            assertThat(state.snackbarMessage).isNull()
+            snackbarDispatcher.clear()
```

Comment:
Aren't you effectively testing `SnackbarDispatcher` here?
Then this should be done in a `SnackbarDispatcherTest` so that we can avoid retesting it in presenters.

Reply:
It's not really that I'm testing `SnackbarDispatcher` as that I need some way to clear the snackbar message. Previously it was done automatically in the presenter, but now it's done automatically *in the view*, which we don't have in the tests... Without clearing it, the tests are a bit more difficult to follow.

Reply:
Yeah in hindsight makes me think that perhaps it was better clearing it in the presenter, what was the disadvantage?